pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -25,10 +26,51 @@
 from .optimization_utils import find_next_gate
 
 
+def _ops_equal(op1, op2):
+    """Checks if two operators are equal up to class, data, hyperparameters, and wires"""
+    return (
+        op1.__class__ is op2.__class__
+        and (op1.data == op2.data)
+        and (op1.hyperparameters == op2.hyperparameters)
+        and (op1.wires == op2.wires)
+    )
+
+
+def _are_inverses(op1, op2):
+    """Checks if two operators are inverses of each other
+
+    Args:
+        op1 (~.Operator)
+        op2 (~.Operator)
+
+    Returns:
+        Bool
+    """
+    # op1 is self-inverse and the next gate is also op1
+    if op1 in self_inverses and op1.name == op2.name:
+        return True
+
+    # Either gate is the in-place inverse of the other
+    name_set = {op1.name, op2.name}
+    shortest_name = min(name_set)
+    if {shortest_name, shortest_name + ".inv"} == name_set:
+        return True
+
+    # op1 is an `Adjoint` class and its base is equal to op2
+    if isinstance(op1, Adjoint) and _ops_equal(op1.base, op2):
+        return True
+
+    # op2 is an `Adjoint` class and its base is equal to op1
+    if isinstance(op2, Adjoint) and _ops_equal(op2.base, op1):
+        return True
```

Comment:
nice!

Reply:
Not sure if important, but this is an inverse check in the programmatic sense. It will be false for `RX(1), RX(-1)`, no?

Reply:
But they would be combined by the `merge_rotations` transform.

Reply:
It could still be unexpected behaviour, because there is an inverse that is not cancelled. Worth mentioning in the docstring?

Reply:
Outside the scope of this PR, but maybe we can  go back over this transform at some point.